paper.md

```diff
@@ -0,0 +1,144 @@
+character). The documentation comments, written in reST [@rest],
+describe to the reader how the functions, parameters, and variables should be
+used. Running CMinx on the annotated source code generates reST files containing
+the API documentation. The reST files can then be converted into static
+websites with tools such as Sphinx [@sphinx] or
+easily converted to another format via Pandoc [@pandoc].
+
+Unlike other solutions for documenting CMake modules, CMinx knows the CMake
+language's grammar. This enables CMinx to automatically extract function/macro
+signatures (even when functions are not documented). CMinx also integrates
+seamlessly into existing CMake build systems. CMinx's output is highly
+customizable and easily controlled via a YAML [@yaml] configuration file.
+CMinx has already proved to be an invaluable
+productivity tool in the authors' other projects, and, given that other
+scientific software projects also rely heavily on source code written in CMake,
+we anticipate CMinx will prove invaluable to many additional projects as well.
+
+# Statement of need
+
+The exascale-era of high-performance scientific computing arguably started when
+the Frontier supercomputer achieved a performance of more than 1.1 exaFLOPs
+(10$^{18}$ floating-point operations per second) on the High-Performance
+Linpack Benchmark [@frontier]. With the breakdown of Dennard scaling
+(*i.e.*, the power usage of a transistor is proportional to its area),
+achieving exascale performance is presently only possible by relying
+on heterogeneous hardware. Unfortunately, hardware heterogeneity significantly
+complicates the process of developing high-performance scientific software.
+For many high-performance software packages, these complications start in the
+build system.
+
+The process of building a software package written in a compiled language
+(*e.g.*, C or C++) has always been complicated. Over the years, various build
+system solutions have evolved to ease the process. Historically, there
+has been a propensity to treat each build system as a one-off use case. This is
+understandable since build systems have tended to be relatively small (*i.e.*,
+less than ~1K lines of code) and tightly coupled to the identity of the package.
+With build system complexity at an all-time high, there is now an
+increasing desire (at least within scientific computing communities) to treat
+the underlying build system infrastructure as code. This means that the
+build system should be modularized, and those modules should be documented,
+tested, and reusable. With the popularity of C/C++ for high-performance
+computing, "build system" is increasingly becoming synonymous with CMake
+(*N.B.* CMake is often described as a build system generator; however, since
+the typical CMake workflow encapsulates running the build system, the
+present manuscript ignores the "generator" distinction). Therefore with the
```

Comment:
Too specific. This could be moved into a footnote or left out entirely.

Reply:
I dropped it.